src/lib/components/MainPanel.js

```diff
@@ -66,18 +61,12 @@ export default class MainPanel extends React.Component {
     scheduleSnoozedTab(time);
   }
 
-  handleTimeSelect(value) {
-    this.setState({ dateChoice: value });
-  }
-
   closeTimeSelect() {
     this.setState({ datepickerActive: false });
   }
 
-  confirmTimeSelect() {
-    const { dateChoice } = this.state;
+  confirmTimeSelect(dateChoice) {
     const { scheduleSnoozedTab } = this.props;
-
     if (!dateChoice) { return; }
```

Comment:
> So, I was able to set a time in the past by going to tomorrow, picking 9am, then going back to today.

Hmm, I'll have to think how to finesse that exactly. Display a warning? Force the time to now? Just not accept a click on the snooze button until the time is valid?

> (It also looks like setting an alarm for a time in the past doesn't ever fire the trigger, so we should really check for that before setting the alarm.)

Huh, that's weird. [This *should* be catching all tabs with times before the current time](https://github.com/bwinton/SnoozeTabs/blob/master/src/background.js#L65). Though, of course, if we don't have a wake time anytime soon (i.e. because it's in the past), that code will never fire. Maybe the wake timer also needs to be guarded so it never get set for any past time - e.g. `Math.max(snoozeTime, now)` or something